src/wp-includes/functions.php

```diff
@@ -7635,3 +7635,42 @@ function is_php_version_compatible( $required ) {
 function wp_fuzzy_number_match( $expected, $actual, $precision = 1 ) {
 	return abs( (float) $expected - (float) $actual ) <= $precision;
 }
+
+/**
+ * Wraps inline JavaScript in <script> tags.
+ *
+ * Enables to inject attributes in the <script> tag of inline scripts printed through this function.
+ *
+ * @param string	$javascript Inline JavaScript code.
+ * @param array		$attributes Optional. <script> tag attributes.
+ * @param bool		$echo 		Optional. Prints to the page is true, returns the value otherwise.	
+ * @return string|null			Inline JavaScript code wrapped around <script> tags if $echo is true, null otherwise.
+ */
+function inline_js( $javascript, $attributes = array(), $echo = true ) {
+	$attributes_string = '';
+
+	// $attributes entries without key are added to $attributes_string without the `=`,
+	// these elements will not be escaped.
+	foreach ( $attributes as $attribute_name => $attribute_value ) {
+		if ( is_int( $attribute_name )) {
+			if ( is_string( $attribute_value ) && $attribute_value !== '' ) {
+				$attributes_string .= ' ' . $attribute_value;
+			}
+		} else {
+			$attributes_string .= sprintf( " %s='%s'", $attribute_name, esc_attr( $attribute_value ) );
+		}
```

Comment:
It makes sense, but I did it like this because it was more flexible, I can for example add a string in the array that could be  holding 'attribute_name="attribute_value"' or '', for example here: https://github.com/WordPress/wordpress-develop/blob/ba0b758d077c5f67a1f4b4a6ac9b20c7116690e1/src/wp-includes/class.wp-scripts.php#L236-L239 
I agree it is not very clean, maybe I should just replace the array with a string.

Reply:
Here's the version with the changes you suggested: https://github.com/enricocarraro/wordpress-develop/blob/52588d60ffcfcdcdead314a06edadb2b82c05a9c/src/wp-includes/functions.php#L7651-L7663
I add the attribute type if needed, so that all the other parts of wp don't have to check for that.